From fc5e202f585a0b4a11f8e4981895cf3fb3b8dd35 Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl@gmail.com>
Date: Mon, 27 Jan 2020 22:39:31 +0100
Subject: [PATCH] Fix PySide related checks

Signed-off-by: Bernd Waibel <waebbl@gmail.com>
---
 CMakeLists.txt | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 89f9a58..311fd3f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -989,6 +989,14 @@ endif()
             endif()
         endif()
 
+        if(NOT SHIBOKEN_INCLUDE_DIR AND TARGET Shiboken2::libshiboken)
+            get_property(SHIBOKEN_INCLUDE_DIR TARGET Shiboken2::libshiboken PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+        endif(NOT SHIBOKEN_INCLUDE_DIR AND TARGET Shiboken2::libshiboken)
+
+        if(NOT SHIBOKEN_LIBRARY AND TARGET Shiboken2::libshiboken)
+            get_property(SHIBOKEN_LIBRARY TARGET Shiboken2::libshiboken PROPERTY IMPORTED_SONAME_GENTOO)
+        endif(NOT SHIBOKEN_LIBRARY AND TARGET Shiboken2::libshiboken)
+
         if(NOT SHIBOKEN_INCLUDE_DIR)
             MESSAGE("====================\n"
                     "shiboken2 not found.\n"
@@ -996,6 +1004,14 @@ endif()
         endif(NOT SHIBOKEN_INCLUDE_DIR)
 
         find_package(PySide2 QUIET)# REQUIRED
+        if(NOT PYSIDE_INCLUDE_DIR AND TARGET PySide2::pyside2)
+            get_property(PYSIDE_INCLUDE_DIR TARGET PySide2::pyside2 PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+        endif(NOT PYSIDE_INCLUDE_DIR AND TARGET PySide2::pyside2)
+
+        if(NOT PYSIDE_LIBRARY AND TARGET PySide2::pyside2)
+            get_property(PYSIDE_LIBRARY TARGET PySide2::pyside2 PROPERTY IMPORTED_SONAME_GENTOO)
+        endif(NOT PYSIDE_LIBRARY AND TARGET PySide2::pyside2)
+
         if(NOT PYSIDE_INCLUDE_DIR)
             MESSAGE("==================\n"
                     "PySide2 not found.\n"
-- 
2.25.0

