--- a/CMakeLists.txt	2020-12-30 12:55:41.525090520 +0800
+++ b/CMakeLists.txt	2020-12-30 14:48:29.236274368 +0800
@@ -68,7 +68,7 @@
 
 # Enabled C++11 for Freecad 0.17 and later
 IF(FREECAD_VERSION VERSION_GREATER 0.16)
-    set(BUILD_ENABLE_CXX_STD "C++11"  CACHE STRING  "Enable C++ standard")
+    set(BUILD_ENABLE_CXX_STD "C++14"  CACHE STRING  "Enable C++ standard")
     set_property(CACHE BUILD_ENABLE_CXX_STD PROPERTY STRINGS
                  "C++11"
                  "C++14"
