From 4160b6a776b8cb14eaac9f35630fee18b6d1a71d Mon Sep 17 00:00:00 2001
From: Bernd Waibel <waebbl@gmail.com>
Date: Mon, 7 Dec 2020 21:09:48 +0100
Subject: [PATCH] use correct uic and rcc calls

Signed-off-by: Bernd Waibel <waebbl@gmail.com>
---
 cMake/FindPySide2Tools.cmake | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/cMake/FindPySide2Tools.cmake b/cMake/FindPySide2Tools.cmake
index 9c97514..51cb8c0 100644
--- a/cMake/FindPySide2Tools.cmake
+++ b/cMake/FindPySide2Tools.cmake
@@ -40,9 +40,9 @@ MACRO(PYSIDE_WRAP_UI outfiles)
         # Especially on Open Build Service we don't want changing date like
         # pyside2-uic generates in comments at beginning., which is why
         # we follow the tool command with in-place sed.
+        #          COMMAND sed -i "/^# /d" "${outfile}"
         ADD_CUSTOM_COMMAND(OUTPUT ${outfile}
-          COMMAND "${PYSIDE2UICBINARY}" "${infile}" -o "${outfile}"
-          COMMAND sed -i "/^# /d" "${outfile}"
+          COMMAND "${PYSIDE2UICBINARY}" -g python -o "${outfile}" "${infile}"
           MAIN_DEPENDENCY "${infile}"
         )
     endif()
@@ -67,9 +67,9 @@ MACRO(PYSIDE_WRAP_RC outfiles)
         # Especially on Open Build Service we don't want changing date like
         # pyside-rcc generates in comments at beginning, which is why
         # we follow the tool command with in-place sed.
+        #          COMMAND sed -i "/^# /d" "${outfile}"
         ADD_CUSTOM_COMMAND(OUTPUT "${outfile}"
-          COMMAND "${PYSIDE2RCCBINARY}" "${infile}" ${PY_ATTRIBUTE} -o "${outfile}"
-          COMMAND sed -i "/^# /d" "${outfile}"
+          COMMAND "${PYSIDE2RCCBINARY}" -g python -o "${outfile}" "${infile}"
           MAIN_DEPENDENCY "${infile}"
         )
     endif()
-- 
2.29.2

