From: Huang Rui <vowstar@gmail.com>
Date: Sun, 9 May 2021 21:04:49 +0800
Subject: [PATCH 1/1] use -std=c++11 to fix compile problem

gcc-11 changed default cpp std

Signed-off-by: Huang Rui <vowstar@gmail.com>

--- a/configure.ac	2016-12-21 06:35:08.000000000 +0800
+++ b/configure.ac	2021-05-10 02:38:22.288655517 +0800
@@ -15,6 +15,7 @@
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 LT_INIT([disable-static win32-dll])
 
+AX_CHECK_COMPILE_FLAG([-std=c++11], [CXXFLAGS="$CXXFLAGS -std=c++11"])
 
 AC_ARG_ENABLE(GUI, AS_HELP_STRING([--disable-GUI], [do not build the GUI]))
 AC_ARG_ENABLE(CLI, AS_HELP_STRING([--disable-CLI], [do not build the CLI]))
