--- a/configure.ac	2013-07-10 19:42:04.000000000 +0800
+++ b/configure.ac	2020-07-18 16:48:31.285318617 +0800
@@ -42,7 +42,7 @@ AC_CHECK_FUNCS([localtime_r])
 AC_CHECK_MEMBERS([struct tm.tm_gmtoff],,,[#include <time.h>])
 dnl sqlite3 is needed for newer ipod models (nano5g), and libplist is needed 
 dnl by libgpod sqlite code
-PKG_CHECK_MODULES(LIBGPOD, glib-2.0 >= 2.16.0 gobject-2.0 sqlite3 libplist >= 1.0 gmodule-2.0)
+PKG_CHECK_MODULES(LIBGPOD, glib-2.0 >= 2.16.0 gobject-2.0 sqlite3 libplist-2.0 >= 1.0 gmodule-2.0)
 
 dnl ***************************************************
 dnl The following functions are only available starting