diff --git a/external/CMakeLists.txt b/external/CMakeLists.txt
index 97e0e79c..255e0ba0 100644
--- a/external/CMakeLists.txt
+++ b/external/CMakeLists.txt
@@ -27,12 +27,13 @@ if((SLANG_INCLUDE_PYLIB OR BUILD_SHARED_LIBS) AND NOT fmt_FOUND)
   set_target_properties(fmt PROPERTIES POSITION_INDEPENDENT_CODE ON)
 endif()
 
-FetchContent_Declare(
-  unordered_dense
-  GIT_REPOSITORY https://github.com/martinus/unordered_dense.git
-  GIT_TAG v2.0.0
-  GIT_SHALLOW ON)
-FetchContent_MakeAvailable(unordered_dense)
+# FetchContent_Declare(
+#   unordered_dense
+#   GIT_REPOSITORY https://github.com/martinus/unordered_dense.git
+#   GIT_TAG v2.0.0
+#   GIT_SHALLOW ON)
+# FetchContent_MakeAvailable(unordered_dense)
+find_package(unordered_dense CONFIG REQUIRED)
 
 if(SLANG_INCLUDE_INSTALL)
   install(
@@ -44,10 +45,10 @@ if(SLANG_INCLUDE_INSTALL)
           ${PROJECT_SOURCE_DIR}/external/span.hpp
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
     COMPONENT slang_Development)
-  install(
-    DIRECTORY ${unordered_dense_SOURCE_DIR}/include/ankerl
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-    COMPONENT slang_Development)
+#  install(
+#    DIRECTORY ${unordered_dense_SOURCE_DIR}/include/ankerl
+#    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+#    COMPONENT slang_Development)
 
   if(NOT fmt_FOUND)
     install(
@@ -59,5 +60,5 @@ if(SLANG_INCLUDE_INSTALL)
       PRIVATE_HEADER EXCLUDE_FROM_ALL)
   endif()
 
-  install(TARGETS unordered_dense EXPORT slangTargets)
+  # install(TARGETS unordered_dense EXPORT slangTargets)
 endif()
