From: Huang Rui <vowstar@gmail.com>
Date: Wed, 20 Oct 2021 20:00:09 +0800
Subject: [PATCH] fix FindOCC.cmake for gentoo build

diff -urN a/CMakeModules/FindOCC.cmake b/CMakeModules/FindOCC.cmake
--- a/CMakeModules/FindOCC.cmake	2021-10-20 20:12:51.838258419 +0800
+++ b/CMakeModules/FindOCC.cmake	2021-10-20 20:13:05.742323270 +0800
@@ -113,11 +113,13 @@
     )
   endif()
 else(WIN32)
+  FILE(GLOB OCC_SEARCH_PATHS "/usr/include/opencascade*" "/usr/local/include/opencascade*")
   FIND_PATH(OCC_INCLUDE_DIR Standard_Version.hxx
     /usr/include/opencascade
     /usr/local/include/opencascade
     /opt/opencascade/include
     /opt/opencascade/inc
+    PATHS ${OCC_SEARCH_PATHS}
   )
   FIND_LIBRARY(OCC_LIBRARY TKernel
     HINTS
