From: Huang Rui <vowstar@gmail.com>
Date: Wed, 20 Oct 2021 20:00:09 +0800
Subject: [PATCH] fix FindOCC.cmake for gentoo build

diff -urN a/CMakeModules/FindOCC.cmake b/CMakeModules/FindOCC.cmake
--- a/CMakeModules/FindOCC.cmake	2021-10-20 20:12:51.838258419 +0800
+++ b/CMakeModules/FindOCC.cmake	2021-10-20 20:13:05.742323270 +0800
@@ -113,19 +113,29 @@
     )
   endif()
 else(WIN32)
-  FIND_PATH(OCC_INCLUDE_DIR Standard_Version.hxx
+  FILE(GLOB OCC_INC_SEARCH_PATHS
+    /usr/include/opencascade*
+    /usr/local/include/opencascade*
     /usr/include/opencascade
     /usr/local/include/opencascade
     /opt/opencascade/include
     /opt/opencascade/inc
   )
-  FIND_LIBRARY(OCC_LIBRARY TKernel
-    HINTS
+  FILE(GLOB OCC_LIB_SEARCH_PATHS
+    /usr/lib64/opencascade*
+    /usr/lib64/opencascade
     ${OCC_LIBRARY_DIR}
     /usr
     /usr/local
     /opt/opencascade
     /opt/opencascade/lin64/gcc
+  )
+  FIND_PATH(OCC_INCLUDE_DIR Standard_Version.hxx
+    PATHS ${OCC_INC_SEARCH_PATHS}
+  )
+  FIND_LIBRARY(OCC_LIBRARY TKernel
+    HINTS
+    PATHS ${OCC_LIB_SEARCH_PATHS}
     PATH_SUFFIXES lib
   )
 endif(WIN32)
