--- modules/dnn/misc/tflite/schema_generated.h	2023-06-28 19:53:33.000000000 +0800
+++ modules/dnn/misc/tflite/schema_generated.h.new	2023-08-24 15:01:25.221486781 +0800
@@ -10,7 +10,7 @@
 // generated, otherwise it may not be compatible.
 static_assert(FLATBUFFERS_VERSION_MAJOR == 23 &&
               FLATBUFFERS_VERSION_MINOR == 5 &&
-              FLATBUFFERS_VERSION_REVISION == 9,
+              FLATBUFFERS_VERSION_REVISION == FLATBUFFERS_VERSION_REVISION,
              "Non-compatible flatbuffers version included");
 
 namespace opencv_tflite {
