--- a/CMakeLists.txt	2023-09-10 10:10:39.792343225 -0400
+++ b/CMakeLists.txt	2023-09-10 10:11:02.692067746 -0400
@@ -215,12 +215,7 @@
 # Look for Python. We prefer some variant of python 3 if the system has it.
 FIND_PACKAGE(PythonInterp 3 QUIET)
 if (PythonInterp_FOUND)
-  # pyverbs can only use python3:
-  if (NO_PYVERBS)
     set(CYTHON_EXECUTABLE "")
-  else()
-    FIND_PACKAGE(cython)
-  endif()
 else()
   # But we still must have python (be it 2) for the build process:
   FIND_PACKAGE(PythonInterp REQUIRED)
--- a/pyverbs/pd.pyx	2023-09-10 10:11:32.278378532 -0400
+++ b/pyverbs/pd.pyx	2023-09-10 10:12:16.831176061 -0400
@@ -152,7 +152,7 @@
 
 
 cdef void *pd_alloc(v.ibv_pd *pd, void *pd_context, size_t size,
-                  size_t alignment, v.uint64_t resource_type):
+                  size_t alignment, v.uint64_t resource_type) noexcept:
     """
     Parent Domain allocator wrapper. This function is used to wrap a
     user-defined Python alloc function which should be a part of pd_context.
@@ -174,7 +174,7 @@
 
 
 cdef void pd_free(v.ibv_pd *pd, void *pd_context, void *ptr,
-                     v.uint64_t resource_type):
+                     v.uint64_t resource_type) noexcept:
     """
     Parent Domain deallocator wrapper. This function is used to wrap a
     user-defined Python free function which should be part of pd_context.
